File: 240206/최고의 13위치 2/best-place-of-13-2.py
n=int(input())

grid=[list(map(int,input().split())) for _ in range(n)]

ans=0
for i in range(n):
    for j in range(n-2):
        for k in range(n):
            for l in range(n-2):
                sum=0
                if i==k and (j==l or j+1==l or j+2==l) :
                    continue
                sum+=(grid[i][j]+grid[i][j+1]+grid[i][j+2]+\
                grid[k][l]+grid[k][l+1]+grid[k][l+2])
                ans=max(ans,sum)


print(ans)

File: 260109/최고의 13위치 2/best-place-of-13-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const grid = input.slice(1, n + 1).map(line => line.split(' ').map(Number));

// Please write your code here.

let ans  = 0;

for(let i=0; i<n; i++){
    for(let j=0; j<n-2; j++){
        for(let k=0; k<n; k++){
            for(let l=0; l<n-2; l++){
                if(i===k && j+3>l){
                    continue;
                }
                let cnt = 0;
                cnt+= (grid[i][j] + grid[i][j+1] + grid[i][j+2]);
                cnt+= (grid[k][l] + grid[k][l+1] + grid[k][l+2]);
                ans = Math.max(ans,cnt);

            }
        }
    }
}

console.log(ans)